pkg/framework/objects/environment/resource.go

```
@@ -0,0 +1,259 @@
+		envToUpdate.Use_Custom_Branch = plan.UseCustomBranch.ValueBool()
+	} else {
+		envToUpdate.Use_Custom_Branch = state.UseCustomBranch.ValueBool()
+	}
+
+	if plan.CustomBranch.ValueString() != state.CustomBranch.ValueString() {
+		customBranch := plan.CustomBranch.ValueString()
+		envToUpdate.Custom_Branch = &customBranch
+	} else {
+		customBranch := state.CustomBranch.ValueString()
+		envToUpdate.Custom_Branch = &customBranch
+	}
+
+	if plan.DeploymentType.ValueString() != state.DeploymentType.ValueString() {
+		deploymentType := plan.DeploymentType.ValueString()
+		envToUpdate.DeploymentType = &deploymentType
+	} else {
+		deploymentType := state.DeploymentType.ValueString()
+		envToUpdate.DeploymentType = &deploymentType
+	}
+
+	if plan.ExtendedAttributesID.ValueInt64() != state.ExtendedAttributesID.ValueInt64() {
+		extendedAttrID := int(plan.ExtendedAttributesID.ValueInt64())
+		envToUpdate.ExtendedAttributesID = &extendedAttrID
+	} else {
+		extendedAttrID := int(state.ExtendedAttributesID.ValueInt64())
+		envToUpdate.ExtendedAttributesID = &extendedAttrID
+	}
+
+	if plan.ConnectionID.ValueInt64() != state.ConnectionID.ValueInt64() {
+		connID := int(plan.ConnectionID.ValueInt64())
+		envToUpdate.ConnectionID = &connID
+	} else {
+		connID := int(state.ConnectionID.ValueInt64())
+		envToUpdate.ConnectionID = &connID
+	}
+
+	if plan.EnableModelQueryHistory.ValueBool() != state.EnableModelQueryHistory.ValueBool() {
+		envToUpdate.EnableModelQueryHistory = plan.EnableModelQueryHistory.ValueBool()
+	} else {
+		envToUpdate.EnableModelQueryHistory = state.EnableModelQueryHistory.ValueBool()
+	}
+
+	_, err := r.client.UpdateEnvironment(
+		int(state.ProjectID.ValueInt64()),
+		int(state.EnvironmentID.ValueInt64()),
+		envToUpdate,
+	)
+	if err != nil {
+		resp.Diagnostics.AddError("Error updating environment", err.Error())
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
```

Bit of a nitpick, but I'd say we read the environment again and apply _that_ to the state, not our plan. Low chance of this happening, but if the update method fails silently for whatever reason, the state won't reflect reality post-update.